src/Nethermind/Nethermind.Core/Timers/FunctionalTimer.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nethermind.Core.Timers;
+
+public static class FunctionalTimer
```

Comment:
why not use [`PeriodicTimer`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.periodictimer?view=net-6.0) ?

Reply:
I like this interface better. Call a function, thats it. No need to store the timer somewhere, or remember to dispose it.

Reply:
`PeriodicTimer` can be a local variable, no need to store it like here:

https://github.com/NethermindEth/nethermind/blob/c977179ce2176d36248083ff73ba15794b5d0d54/src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs#L120-L141

Though we should add `using` there.

Reply:
But you still need a variable, you still need a custom loop.